src/apps/dialogs/initializers/LanguageSelectDialog.js

```diff
@@ -1,87 +1,49 @@
-// eslint-disable-next-line import/no-unresolved
-import { TJSDialog } from '@typhonjs-fvtt/runtime/svelte/application';
-
-import isCustomLanguage from '../../utils/isCustomLanguage';
-import LanguageSelectDialogComponent from '../LanguageSelectDialog.svelte';
+import TraitSelectDialog from './TraitSelectDialog';
 
 /**
  * Provides a dialog for creating documents that by default is modal and not draggable.
  */
-export default class LanguageSelectDialog extends TJSDialog {
-  _selected = [];
-
-  _disabled = [];
-
-  constructor(options) {
-    super({
-      title: 'Select Languages',
-      content: {
-        class: LanguageSelectDialogComponent,
-        props: {
-        }
-      }
-    }, { classes: ['a5e-sheet'], width: 480 });
-
-    if (Array.isArray(options?.languages)) this._selected = options.languages;
-    if (Array.isArray(options?.disabled)) this._disabled = options.disabled;
-
-    this.promise = new Promise((resolve) => {
-      this.resolve = resolve;
-    });
+export default class LanguageSelectDialog extends TraitSelectDialog {
+  /**
+   * @param {string} titleName Name to include in the title
+   * @param {*} options See TraitSelectDialog options
+   */
+  constructor(titleName, options) {
+    super(
+      game.i18n.format('A5E.LanguagesConfigurationPrompt', titleName),
+      [{
+        label: 'A5E.Languages',
+        traits: CONFIG.A5E.languages
+      },
+      {
+        label: 'Other Languages'
+      }],
+      options
+    );
   }
 
   /**
-   *
-   * @param {[:string]} selected A list of languages to preselect
-   * @param {[:string]} disabled A list of languages which to disable
-   * @returns
+   * @param {string} Name to include in the title
+   * @param {[string]]} known List of known languages, users can not change these.
+   * @param {[string]} recommended List of recommended languages, users may change these.
+   * @param {number=0} additional
+   * @return Returns the new list of languages, or `known` if user doesn't submit their choices
    */
-  getLanguages() {
-    const core = [];
-
-    Object.entries(CONFIG.A5E.languages).forEach((element) => {
-      core.push({
-        value: element[0],
-        name: game.i18n.localize(element[1]),
-        selected: this._selected.includes(element[0]),
-        disabled: this._disabled.includes(element[0])
-      });
-    });
-
-    const custom = [];
-    this._selected.forEach((lang) => {
-      if (isCustomLanguage(lang)) {
-        custom.push(lang);
+  static async createRecommendLanguages(titleName, known, recommended, additional = 0) {
```

Comment:
Wondering if we can come up with a better name for the function. It's a bit hard to guess what this does when calling it from outside the file.